src/blocks/DetailsBlock/DetailsBlock.astro

```diff
@@ -0,0 +1,34 @@
+---
+import type { DetailsBlockFragment } from '@lib/types/datocms';
+
+interface Props {
+    block: DetailsBlockFragment;
+}
+
+const { block } = Astro.props;
+const splitCharacter = '|';
+---
+
+<dl data-layout={block.layout}>
+  {
+    Object.entries(block.details).map(([key, value]) => (
+      <>
+        <dt>{key}</dt>
+        {(value as string).split(splitCharacter).map((splitValue) => (
+          <dd>{splitValue}</dd>
```

Comment:
I would also trim the value, so in the CMS editors can write "a | b | c" instead of "a|b|c", so `splitValue.trim()`